lib/widgets/emoji.dart

```diff
@@ -71,7 +71,9 @@ class UnicodeEmojiWidget extends StatelessWidget {
           SizedBox(height: boxSize, width: boxSize),
           PositionedDirectional(start: 0, child: Text(
             textScaler: textScaler,
-            style: TextStyle(fontSize: size),
+            style: TextStyle(
+              fontFamily: 'Apple Color Emoji',
```

Comment:
emoji: Use "Apple Color Emoji" font on iOS/macOS for UnicodeEmojiWidget
    
    Some unicode characters, like U+2764 (❤) or U+00AE (®) can
    have glyphs in non-Emoji fonts, resulting in incorrect
    rendering of such characters, where we specifically want an
    emoji to be displayed.
    
    So, explicitly mention "Apple Color Emoji" to be the font used on
    iOS/macOS for displaying the unicode emoji.

Interesting — I reproduce the issue.

I think the right fix is actually somewhat broader. I'll file an issue and send a PR.

Reply:
Well, filed #1104. Then the broad fix I had in mind turned out not to work: https://github.com/zulip/zulip-flutter/issues/1104#issuecomment-2518782312

I have a draft branch that will fix that issue for emoji in message content, which is the symptom I described in the issue. But we'll still need exactly this fix for this UnicodeEmojiWidget class, found in reaction chips and autocomplete results. So go ahead and leave this commit in your PR; I might also cherry-pick it into the PR I send for #1104.

Reply:
OK, sent #1108. It includes a cherry-pick of this commit — the only change I made to the diff was to add a test.